index.html

```diff
@@ -787,6 +787,48 @@ <h4>`ThingLink`</h4>
                     </p>
                 </section>
             </section>
+            <section id="exploration-secboot" class="normative">
+                <h3>Security Bootstrapping</h3>
+                <p>The purpose of an exploration service is to serve TDs, but only after suitable
+                authentication, and only to authorized parties.  However, since we don't yet have
+                a TD when accessing an exploration service, we can't depend on the security metadata
+                held in TDs to know what kind of authentication and authorization information is needed.
+                Instead, we have to depend on existing security negotation support in HTTP to
+                bootstrap access, at least to the TD itself.
```

Comment:
If TDs adds support for associating security definitions with links, we can Thing Links to split private and public TD parts.

Reply:
If I've understood you correctly then splitting public and private TD parts into separate Thing Link and Thing Description resources would require that every (authenticated) directory actually has to serve two separate resources for every device - a public Thing Link and a private Thing Description. It may also need to serve both a public Thing Link and a private Thing Description for itself. How would that all fit into the prescriptive URL structure of the Directory Service API?

My original suggestion in #135 was that if a client requests a Thing Description without being authenticated, the server could initially respond with a partial TD which only contains security metadata (that could be the body of a 401 response). It could then re-request the same URL once authenticated. However, I think you pointed out that the security metadata in the Thing Description describes its affordances, not the TD itself.

Does the security metadata in a Thing Link describe the security requirements of accessing its links? I seem to recall that links are excluded from the scope of the security member, but I can't find text in the Thing Description specification which says that.

In order to avoid the need for two resources for every device in a directory, could an un-authenticated request on the URL of a Thing Description respond with a 401 response and contain a Thing Link in its body, linking to itself?!

```json
{
    "@context": [
        "https://www.w3.org/2022/wot/td/v1.1",
        "https://w3c.github.io/wot-discovery/context/discovery-context.jsonld"
    ],
    "@type": "ThingLink",
    "id": "https://example.webthings.io/things/mything1",
    "links": [{
        "rel": "self",
        "href": "https://example.webthings.io/things/mything1",
        "type": "application/td+json"
    }],
    "securityDefinitions": {
        "oauth2_sc": {
            "scheme": "oauth2",
            "flow": "code",
            "authorization": "https://example.webthings.io/oauth/authorize",
            "token": "https://example.webthings.io/oauth/token",
            "scopes": [
                "/things/mything1:readwrite",
                "/things/mything1",
                "/things:readwrite",
                "/things"
            ]
        }
    },
    "security": "oauth2_sc"
}
```

If a `WWW-Authentication` header turns out to not contain sufficient metadata, the only other alternative I can think of is to define new semantics in the Thing Description specification for describing the security properties of the Thing Description itself, and it's probably too late for that.

Reply:
I think serving partial TDs is basically a new security mechanism to negotiate access.  I'd rather not do that, since there is a perfectly good vetted negotiation mechanism in HTTP, which already has all kinds of support in libraries, etc.  Since (for now anyway) we are limiting ourselves to HTTP, this PR suggest that we simply use that.